front/src/pages/settings/api/SettingsNewApi.tsx

```diff
@@ -0,0 +1,96 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { DateTime } from 'luxon';
+
+import { SaveAndCancelButtons } from '@/settings/components/SaveAndCancelButtons/SaveAndCancelButtons';
+import { SettingsHeaderContainer } from '@/settings/components/SettingsHeaderContainer';
+import { SettingsPageContainer } from '@/settings/components/SettingsPageContainer';
+import { ExpirationDates } from '@/settings/developers/constants/expirationDates';
+import { IconSettings } from '@/ui/display/icon';
+import { H2Title } from '@/ui/display/typography/components/H2Title';
+import { Select } from '@/ui/input/components/Select';
+import { TextInput } from '@/ui/input/components/TextInput';
+import { SubMenuTopBarContainer } from '@/ui/layout/page/SubMenuTopBarContainer';
+import { Section } from '@/ui/layout/section/components/Section';
+import { Breadcrumb } from '@/ui/navigation/bread-crumb/components/Breadcrumb';
+import { useInsertOneApiKeyMutation } from '~/generated/graphql';
+
+export const SettingsNewApi = () => {
+  const [insertOneApiKey] = useInsertOneApiKeyMutation();
+  const navigate = useNavigate();
+  const [formValues, setFormValues] = useState<{
+    name: string;
+    expirationDate: number;
+  }>({
+    expirationDate: ExpirationDates[0].value,
+    name: '',
+  });
+  const onSave = async () => {
+    const apiKey = await insertOneApiKey({
+      variables: {
+        data: {
+          name: formValues.name,
+          expiresAt: DateTime.now()
+            .plus({ days: formValues.expirationDate })
+            .toISODate(),
+        },
+      },
+    });
+    navigate(`/settings/apis/${apiKey.data?.createOneApiKey?.id}`, {
+      state: apiKey.data?.createOneApiKey?.token,
```

Comment:
that's a new pattern introducing a new state. can we get rid of it and on the detail page fetch from db the api key. 
For performance issue, we might want to avoid doing an extra call on the detail page but this is solved using optimistic rendering on apollo side, we can put that in place later

Reply:
We don't store apiKeys in the db so I cannot fetch them from the db
Need to think about a proper solution there

Reply:
Mmmh I see, even we don't store it can we have the back return it? (re-compute it everytime)
We have already done that for supportUserHash for example

Reply:
I called Lucas, we think using recoil is the a good solution